In [15]:
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sklearn.pipeline import make_pipeline
from sklearn.metrics import mean_squared_error

np.random.seed(42)

x= np.linspace(0, 10, 40).reshape(-1,1)    #0,500,1000

y = 2 + 0.5 * x[:, 0] ** 2 + np.random.normal(0, 5, 40)

x_train, x_test, y_train, y_test = train_test_split(
    x, y, test_size=0.30, random_state = 42             
)

degrees = [1,2,10]           #1,2,3,4,5,10

for degree in degrees:
    model = make_pipeline(
        PolynomialFeatures(degree=degree, include_bias=False),          # bias =True
        LinearRegression()
    )
    model.fit(x_train, y_train)
    train_prediction = model.predict(x_train)
    test_prediction = model.predict(x_test)
    
    train_mse = mean_squared_error(y_train, train_prediction)
    test_mse = mean_squared_error(y_test, test_prediction)
    
    print(f"Polynomial degree:{degree}")
    print(f"Training MSE:{train_mse:.2f}")
    print(f"Testing MSE:{test_mse:.2f}")
    print()


Polynomial degree:1
Training MSE:42.90
Testing MSE:42.99

Polynomial degree:2
Training MSE:20.44
Testing MSE:19.79

Polynomial degree:10
Training MSE:18.26
Testing MSE:38.42



test size 20:
Polynomial degree:1
Training MSE:44.19
Testing MSE:41.82

Polynomial degree:2
Training MSE:20.07
Testing MSE:21.48

Polynomial degree:10
Training MSE:18.94
Testing MSE:15.78

test size 25:
Polynomial degree:1
Training MSE:40.17
Testing MSE:51.32

Polynomial degree:2
Training MSE:19.63
Testing MSE:22.09

Polynomial degree:10
Training MSE:17.45
Testing MSE:43.67

test size 30:
Polynomial degree:1
Training MSE:42.90
Testing MSE:42.99

Polynomial degree:2
Training MSE:20.44
Testing MSE:19.79

Polynomial degree:10
Training MSE:18.26
Testing MSE:38.42